### PR TITLE
improved Horus rotary encoder driver and generic behavior

### LINKED ISSUE #5492: Scroll wheel issue
I have a problem with the scroll wheel on my Horus X12S, it's working erratically in Open-TX, if I scroll down, it will scroll down than jump up, than down again, the same if I scroll UP, it will go UP than down, than UP again.

With FrOS this is not happening, I have reinstalled Open-TX 2.2.0 3 times now, this is still happening, very annoying!

### PR DESCRIPTION
**Pull request by raphaelcoeffic**
Heavily tested on X10 and X7.

fixes #5492

### CODE REVIEW

**Comment by raphaelcoeffic** [comment]
To be able test properly the *before* and *after* state, I found useful to use the following traces:
```
diff --git a/radio/src/opentx.cpp b/radio/src/opentx.cpp
index 54a38340a..d85ef72d1 100644
--- a/radio/src/opentx.cpp
+++ b/radio/src/opentx.cpp
@@ -140,6 +140,8 @@ void per10ms()
     if (scrollRE) {
       rePreviousValue = reNewValue;
       putEvent(scrollRE < 0 ? EVT_ROTARY_LEFT : EVT_ROTARY_RIGHT);
+      if (scrollRE < 0) TRACE("-1");
+      else TRACE("+1");
     }
 #if defined(CPUARM)
     // rotary encoder navigation speed (acceleration) detection/calculation
```

File: radio/src/opentx.cpp [hunk 1]
```diff
@@ -136,7 +136,7 @@ void per10ms()
   if (IS_ROTARY_ENCODER_NAVIGATION_ENABLE()) {
     static rotenc_t rePreviousValue;
     rotenc_t reNewValue = (ROTARY_ENCODER_NAVIGATION_VALUE / ROTARY_ENCODER_GRANULARITY);
-    int8_t scrollRE = reNewValue - rePreviousValue;
+    rotenc_t scrollRE = reNewValue - rePreviousValue;
```
Comment (raphaelcoeffic) [inline radio/src/opentx.cpp]: Please note that this one is probably not necessary (works either way), but I found it cleaner to use the same type here.

Comment (bsongis) [reply radio/src/opentx.cpp]: TODO: check if we use this code on a 8bits platform (we are in an interrupt) and in this case what is rotenc_t type

**Comment by raphaelcoeffic** [comment]
@bsongis I'm testing a way better code right now (using state table), as I could still see some glitches while further testing.

**Comment by bsongis** [comment]
no rush, we will keep it for 2.2.2

File: radio/src/opentx.cpp (hunk 1 above)
Comment (raphaelcoeffic) [reply radio/src/opentx.cpp]: There is a define for that, it's basically 8bit an anything but ARM.

**Comment by raphaelcoeffic** [comment]
exactly, no rush, this time I want it to be as perfect as possible ;-)

**Comment by raphaelcoeffic** [comment]
@bsongis ok, now it's really rock solid. Works 100% for me on X10 and X7.

**Comment by raphaelcoeffic** [comment]
[X12S_rot_enc_fix.zip](https://github.com/opentx/opentx/files/1564645/X12S_rot_enc_fix.zip)

**Comment by raphaelcoeffic** [comment]
[X10_rot_enc_fix.zip](https://github.com/opentx/opentx/files/1564654/X10_rot_enc_fix.zip)

**Comment by bsongis** [comment]
We will need more tests from nightly builds, but this seems good

**Pull request merged**